plugins/modules/digital_ocean_floating_ip.py

```diff
@@ -287,6 +287,21 @@ def create_floating_ips(module, rest):
     if module.params['droplet_id'] is not None:
         payload["droplet_id"] = module.params['droplet_id']
 
+    # Get existing floating IPs
+    response = rest.get('floating_ips/')
+    status_code = response.status_code
+    json_data = response.json
+
+    # Exit unchanged if any of them are assigned to this Droplet already
+    if status_code == 200:
+        for floating_ip in json_data.get('floating_ips', []):
+            droplet = floating_ip.get('droplet', None)
+            if droplet is not None:
+                droplet_id = droplet.get('id', None)
+                if droplet_id is not None:
+                    if droplet_id == module.params['droplet_id']:
+                        module.exit_json(changed=False, data={'floating_ip': floating_ip['ip']})
```

Comment:
Consider `floating_ip.get('ip', None)` here.

Reply:
Cool, I think this is in better shape now.